### PR TITLE
API: create can handle a form submission now

### PR DESCRIPTION
**Pull request by gaborcsardi**
Plus:
* do a 302 redirect after the notebook is created
* show an error page if user is not logged in

### CODE REVIEW

File: htdocs/api.R [hunk 1]
```diff
@@ -55,3 +56,25 @@ run <- function(url, query, body, headers) {
...
+    <link rel="stylesheet" type="text/css" href="/css/rcloud.css"/>
+  </head>
+  <body id="goodbye">
+    <div class="navbar navbar-inverse navbar-fixed-top">
+      <div class="navbar-header">
+        <a class="navbar-brand" href="/login.R">RCloud</a>
+      </div>
+    </div>
+    <div class="container" id="main-div" style="margin-top: 100px;">
+      <p>To create an RCLoud notebook you need to be logged in, but
```
Comment (shaneporter) [inline htdocs/api.R]: RCloud

Comment (gaborcsardi) [inline htdocs/api.R]: Thanks!

**Comment by gordonwoodhull** [comment]
I'm going to leave this open until I see it work - I guess the problems with RStudio Export happen earlier in the chain, but I figure it's easier just to keep the branch/PR open.

We also need to star and show the notebook, something I neglected in my initial implementation. By default notebooks are not visible to other users, and we star them so that they'll show up in what passes for current notebooks in the tree. The star design is somewhat disputable (#2302) but that's what we've got for now.

Usually we do this [from the javascript side](https://github.com/att/rcloud/blob/7d562735613c2bb2a1e7b7ce0720a390226cb748/htdocs/editor_tab.js#L1394-L1395) but they are simple APIs on the R side too.

**Comment by gaborcsardi** [comment]
OK, I do the store-token-redirect-pickup loop now if the user is not logged in. Seems to work fine.

I'll do the staring in a minute.

**Comment by gaborcsardi** [comment]
@gordonwoodhull I am not sure what `RSclient::RS.eval.qap(c, as.call(list(caps$rcloud$create_notebook, content)))` returns, but apparently I cannot use that to star the notebook.....

How do I star the new notebook?

**Comment by gordonwoodhull** [comment]
`caps$rcloud` should contain all the authenticated OCAPS, [which are listed here](https://github.com/att/rcloud/blob/develop/rcloud.support/R/ocaps.R). I don't know what that `RSClient` stuff does TBH but I think if you repeat the pattern you should be able to call any of them.

**Comment by gordonwoodhull** [comment]
Er, looks like there was some careless editing in ocaps.R, the stars function [are actually in the unauthenticated ocaps](https://github.com/att/rcloud/blob/35cf4469beed930f91fcada16966f2fdbac0e25c/rcloud.support/R/ocaps.R#L162), which the authenticated ocaps inherit.

**Comment by gordonwoodhull** [comment]
Okay, I guess I'll take the create/show bits. 

It occurred to me that we should really have an `api.R/loggedin` boolean `GET` rather than pushing the data and stashing it. So I guess I will document `api.R` in the wiki with a big warning that it's subject to change.

**Comment by gaborcsardi** [comment]
YEah, we don't have to advertise API imo, it is not a real API, anyway.

I am not sure how much `api.R/loggedin` would help, because I cannot query it without CORS.

**Comment by gordonwoodhull** [comment]
Inevitably everything becomes an API over time. 😜 

That's a good point about same-origin problems. It's probably not good to rely on cookies for a REST API, and we'll need to pass some token as a parameter instead.

**Comment by gaborcsardi** [comment]
Exactly. Another reason I would not call the current code an API just yet.

As I see it, the current form submission is a hack, and a proper REST api should use tokens. But that would make the RStudio addin somewhat inconvenient.

**Comment by gaborcsardi** [comment]
Btw. in theory we could also read out the cookie from the local browser I guess. See e.g. https://gist.github.com/dacort/bd6a5116224c594b14db (I haven't tried this.)

**Comment by gordonwoodhull** [comment]
Thanks @gaborcsardi, we'll leave these clues for later. 

Here is the anti-documentation (feel free to edit if inaccurate): 
https://github.com/att/rcloud/wiki/api.R

**Comment by gaborcsardi** [comment]
OK, looks good to me. :)

**Comment by gordonwoodhull** [comment]
Thanks @gaborcsardi, rebased and merged.

**Pull request closed without merging**